pyuvdata/parameter.py

```diff
@@ -192,6 +176,34 @@ def __eq__(self, other):
                 if self.value.shape != other.value.shape:
                     print(f"{self.name} parameter value is array, shapes are different")
                     return False
+
+                # check to see if strict types are used
+                if self.strict_type:
+                    # types must match
+                    if other.strict_type:
+                        # both strict, expected_type must match
+                        if self.expected_type != other.expected_type:
+                            print(
+                                f"{self.name} parameter has incompatible types. "
+                                f"Left is {self.expected_type}, right is "
+                                f"{other.expected_type}"
+                            )
+                            return False
+                    elif not isinstance(self.value.item(0), other.expected_type):
+                        print(
+                            f"{self.name} parameter has incompatible dtypes. Left is "
+                            f"{self.value.dtype}, right is {other.expected_type}"
+                        )
+                        return False
+                elif other.strict_type:
+                    # types must match in the other direction
+                    if not isinstance(other.value.item(0), self.expected_type):
+                        print(
+                            f"{self.name} parameter has incompatible dtypes. Left is "
+                            f"{other.value.dtype}, right is {other.expected_type}"
```

Comment:
We're using `other` for both left and right here. Should right be `self.expected_type`?